pkg/cmd/pipeline/translate.go

```diff
@@ -0,0 +1,226 @@
+var maxLogOutputRaw = int(math.Floor(float64(maxLogOutput)*3) / 4)
+
+//CfgData represents the data to be include in a configmap
+type CfgData struct {
+	Name       string
+	Namespace  string
+	Status     string
+	Output     string
+	Repository string
+	Branch     string
+	Filename   string
+	Manifest   []byte
+	Icon       string
+}
+
+// TranslateConfigMap translates the app into a configMap
+func TranslateConfigMap(ctx context.Context, data *CfgData, c kubernetes.Interface) (*apiv1.ConfigMap, error) {
+	cmap, err := configmaps.Get(ctx, TranslatePipelineName(data.Name), data.Namespace, c)
+	if err != nil {
+		if !oktetoErrors.IsNotFound(err) {
+			return nil, err
+		}
+		cmap = translatedConfigMapSandBox(data)
+		err := configmaps.Create(ctx, cmap, cmap.Namespace, c)
+		if err != nil {
+			if k8sErrors.IsAlreadyExists(err) {
+				return nil, fmt.Errorf("A new deploy operation has started")
+			}
+			return nil, err
+		}
+	}
+
+	if err := updateCmap(cmap, data); err != nil {
+		return nil, err
+	}
+	if err := configmaps.Deploy(ctx, cmap, cmap.Namespace, c); err != nil {
+		if k8sErrors.IsConflict(err) {
+			return nil, fmt.Errorf("Another pipeline operation is being executed")
+		}
+		return nil, err
+	}
+	return cmap, nil
+}
+
+// SetOutput sets the output of a config map
+func SetOutput(cfg *apiv1.ConfigMap, output string) *apiv1.ConfigMap {
+	cfg.Data[outputField] = base64.StdEncoding.EncodeToString([]byte(output))
+	return cfg
+}
+
+// UpdateConfigMap updates the configmaps fields
+func UpdateConfigMap(ctx context.Context, cmap *apiv1.ConfigMap, data *CfgData, c kubernetes.Interface) error {
+	cmap, err := configmaps.Get(ctx, cmap.Name, cmap.Namespace, c)
+	if err != nil {
+		return err
+	}
+	if err := updateCmap(cmap, data); err != nil {
+		return err
+	}
+	return configmaps.Deploy(ctx, cmap, cmap.Namespace, c)
+}
+
+// TranslatePipelineName translate the name into the pipeline name
+func TranslatePipelineName(name string) string {
+	return fmt.Sprintf("okteto-git-%s", name)
+}
+
+func translateOutput(output *bytes.Buffer) []byte {
+	// If the output is larger than the currentMaxLimit for the logs trim it.
+	// We can't really truncate the buffer since we would end up with an invalid json
+	// line for the last line, so we pick lines from the end while the line fits
+	var data []byte
+	if output.Len() > maxLogOutputRaw {
+		scanner := bufio.NewScanner(output)
+		linesInReverse := []string{}
+		for scanner.Scan() {
+			linesInReverse = append([]string{scanner.Text()}, linesInReverse...)
+		}
+		var cappedOutput string
+		var head string
+		for _, l := range linesInReverse {
+			head = l + "\n"
+			if len(cappedOutput)+len(head) > maxLogOutputRaw {
+				break
+			}
+			cappedOutput = head + cappedOutput
+		}
+		cappedOutput = strings.TrimPrefix(cappedOutput, head)
+		data = []byte(cappedOutput)
+	} else {
+		data = output.Bytes()
+	}
+	return data
+}
+
+func translatedConfigMapSandBox(data *CfgData) *apiv1.ConfigMap {
```

Comment:
A comment explaining what a ConfigMapSandBox and why is needed would be really helpful